lib/views/item-scene.js

```diff
@@ -22,6 +22,49 @@ class ItemScene extends React.Component {
   constructor(props) {
     super(props);
     debug('created');
+
+    let distance = this.props.item.distance;
+    if (typeof distance !== 'number') {
+      distance = null;
+    }
+
+    this.state = {
+      distance,
+    };
+  }
+
+  componentDidMount() {
+    if (this.state.distance !== null ||
+      typeof this.props.item.latitude !== 'number' ||
+      typeof this.props.item.longitude !== 'number') {
+      return;
+    }
+
+    navigator.geolocation.getCurrentPosition((pos) => {
```

Comment:
I tried several parameters but it didn't work on my phone. I'll enquire in a followup.